--- origsrc/PyOpenGL-3.0.0b6/OpenGL/__init__.py	2008-09-07 20:22:17.000000000 -0500
+++ src/PyOpenGL-3.0.0b6/OpenGL/__init__.py	2008-09-22 12:30:17.001362800 -0500
@@ -104,6 +104,7 @@
 PlatformPlugin( 'nt', 'OpenGL.platform.win32.Win32Platform' )
 PlatformPlugin( 'posix ', 'OpenGL.platform.glx.GLXPlatform' )
 PlatformPlugin( 'linux2', 'OpenGL.platform.glx.GLXPlatform' )
+PlatformPlugin( 'cygwin', 'OpenGL.platform.glx.GLXPlatform' )
 PlatformPlugin( 'darwin', 'OpenGL.platform.darwin.DarwinPlatform' )
 
 FormatHandler( 'none', 'OpenGL.arrays.nones.NoneHandler' )
