--- origsrc/PyOpenGL-3.0.0b6/OpenGL/__init__.py	2008-09-07 20:22:17.000000000 -0500
+++ src/PyOpenGL-3.0.0b6/OpenGL/__init__.py	2008-09-22 12:30:17.001362800 -0500
@@ -104,6 +104,7 @@
 PlatformPlugin( 'nt', 'OpenGL.platform.win32.Win32Platform' )
 PlatformPlugin( 'posix ', 'OpenGL.platform.glx.GLXPlatform' )
 PlatformPlugin( 'linux2', 'OpenGL.platform.glx.GLXPlatform' )
+PlatformPlugin( 'cygwin', 'OpenGL.platform.glx.GLXPlatform' )
 PlatformPlugin( 'darwin', 'OpenGL.platform.darwin.DarwinPlatform' )
 
 FormatHandler( 'none', 'OpenGL.arrays.nones.NoneHandler' )
--- origsrc/PyOpenGL-3.0.0b6/OpenGL/platform/glx.py	2008-09-22 12:33:27.360737800 -0500
+++ src/PyOpenGL-3.0.0b6/OpenGL/platform/glx.py	2008-09-22 12:24:19.000000000 -0500
@@ -1,9 +1,20 @@
 """GLX (x-windows)-specific platform features"""
 import ctypes, ctypes.util
+import sys
 from OpenGL.platform import baseplatform, ctypesloader
 
 assert hasattr( ctypes, 'RTLD_GLOBAL' ), """Old ctypes without ability to load .so for global resolution: Get ctypes CVS branch_1_0, not CVS HEAD or released versions!"""
 
+if sys.platform == 'cygwin':
+	libGL = 'cygGL-1.dll'
+	libGLU = 'cygGLU-1.dll'
+	libglut = 'cygglut-3.dll'
+	libgle = 'cyggle-3.dll'
+else:
+	libGL = 'GL'
+	libGLU = 'GLU'
+	libglut = 'glut'
+	libgle = 'gle'
 
 class GLXPlatform( baseplatform.BasePlatform ):
 	"""Posix (Linux, FreeBSD, etceteras) implementation for PyOpenGL"""
@@ -12,12 +23,12 @@
 	# references to GL/GLU functions).
 	GL = OpenGL = ctypesloader.loadLibrary(
 		ctypes.cdll,
-		'GL', 
+		libGL, 
 		mode=ctypes.RTLD_GLOBAL 
 	)
 	GLU = ctypesloader.loadLibrary(
 		ctypes.cdll,
-		'GLU',
+		libGLU,
 		mode=ctypes.RTLD_GLOBAL 
 	)
 	# glut shouldn't need to be global, but just in case a dependent library makes
@@ -25,7 +36,7 @@
 	try:
 		GLUT = ctypesloader.loadLibrary(
 			ctypes.cdll,
-			'glut', 
+			libglut, 
 			mode=ctypes.RTLD_GLOBAL 
 		)
 	except OSError, err:
@@ -38,7 +49,7 @@
 	try:
 		GLE = ctypesloader.loadLibrary(
 			ctypes.cdll,
-			'gle', 
+			libgle, 
 			mode=ctypes.RTLD_GLOBAL 
 		)
 	except OSError, err:
