--- origsrc/PyOpenGL-3.0.0b3/OpenGL/platform/glx.py	2008-03-16 22:13:13.000000000 -0500
+++ src/PyOpenGL-3.0.0b3/OpenGL/platform/glx.py	2008-06-30 02:38:07.453125000 -0500
@@ -1,9 +1,20 @@
 """GLX (x-windows)-specific platform features"""
 import ctypes, ctypes.util
+import sys
 from OpenGL.platform import baseplatform, ctypesloader
 
 assert hasattr( ctypes, 'RTLD_GLOBAL' ), """Old ctypes without ability to load .so for global resolution: Get ctypes CVS branch_1_0, not CVS HEAD or released versions!"""
 
+if sys.platform == 'cygwin':
+	libGL = 'cygGL-1.dll'
+	libGLU = 'cygGLU-1.dll'
+	libglut = 'cygglut-3.dll'
+	libgle = 'cyggle-3.dll'
+else:
+	libGL = 'GL'
+	libGLU = 'GLU'
+	libglut = 'glut'
+	libgle = 'gle'
 
 class GLXPlatform( baseplatform.BasePlatform ):
 	"""Posix (Linux, FreeBSD, etceteras) implementation for PyOpenGL"""
@@ -12,12 +23,12 @@
 	# references to GL/GLU functions).
 	GL = OpenGL = ctypesloader.loadLibrary(
 		ctypes.cdll,
-		'GL', 
+		libGL, 
 		mode=ctypes.RTLD_GLOBAL 
 	)
 	GLU = ctypesloader.loadLibrary(
 		ctypes.cdll,
-		'GLU',
+		libGLU,
 		mode=ctypes.RTLD_GLOBAL 
 	)
 	# glut shouldn't need to be global, but just in case a dependent library makes
@@ -25,7 +36,7 @@
 	try:
 		GLUT = ctypesloader.loadLibrary(
 			ctypes.cdll,
-			'glut', 
+			libglut, 
 			mode=ctypes.RTLD_GLOBAL 
 		)
 	except OSError, err:
@@ -38,7 +49,7 @@
 	try:
 		GLE = ctypesloader.loadLibrary(
 			ctypes.cdll,
-			'gle', 
+			libgle, 
 			mode=ctypes.RTLD_GLOBAL 
 		)
 	except OSError, err:
